### DISCUSSION TITLE
Error connecting to PlanetScale db (Next.js/Vercel)

### DISCUSSION THREAD

**Discussion by jbarrasso** [prisma/prisma#13499] [Q&A]
Hello,

I seem to be having an issue connecting to my planetscale db via Prisma. My Next.js app is deployed on Vercel.

In a local environment, I can read/write to the db just fine. It's only when it's deployed I'm having issues.
I am using a connection string which I have set properly in my Vercel dashboard.
![image](https://user-images.githubusercontent.com/18688696/170431791-a8df7414-db8b-4864-8da6-ba6321641b71.png)

When trying to do a PUT request, it responds with the following error:

`Uncaught (in promise) SyntaxError: Unexpected end of JSON input`
![image](https://user-images.githubusercontent.com/18688696/170429948-48536f9e-4add-40db-a36d-0a367c770597.png)

Frontend:
![image](https://user-images.githubusercontent.com/18688696/170429050-d1d1c0b9-7080-4374-82d0-b9aca1d716d7.png)

Backend:
![image](https://user-images.githubusercontent.com/18688696/170429077-ed1598e7-691b-4075-95b3-368fbd12464c.png)

Below are the headers for the local and deployed environments:
Local:
![image](https://user-images.githubusercontent.com/18688696/170429264-c52141e3-ffdf-4a3e-be62-24c749ff472a.png)

Deployed via Vercel:
![image](https://user-images.githubusercontent.com/18688696/170429298-da1d6d55-24d0-47cd-ad5d-8fcbc7351c46.png)

Anyone have any idea as to what might be going on? I asked this in both the Vercel and PlanetScale forums, just wanted to repost here for extra visibility if anyone has any insight by chance. Thank you!

**Comment by nurul3101** [comment]
Hey @jbarrasso 👋 
Are you getting this error when trying Prisma Data Proxy, does your connection string with `prisma://`?

**Reply by jbarrasso** [reply]
I did set a Prisma Data Proxy up before, but I switched back to a regular connection string planetscale generated for me (mysql://)

**Reply by jbarrasso** [reply]
@nurul3101 

Is it recommended to use data proxy with Planetscale? Can it work without using that feature? Only reason I ask is because I am also using Prisma's Interactive Transactions which can't be used alongside data proxy

**Comment by janpio** [comment]
What error message are you getting on that `/api/users/...` function call on Vercel? You show a network tab where it returns a 404, but not what it outputs (it probably is some HTML, which leads your frontend to be unhappy). Do you have anything helpful in the Vercel functions logs in their UI?

**Reply by jbarrasso** [reply]
@janpio Thanks for the reply - I am not even getting an error in the output logs which is weird. The GET methods work fine on my index.tsx homepage, but nothing is showing for the failed PUT methods for /api/users/[id] which is clearly being called (I can see the failed fetch attempt in chrome)

![image](https://user-images.githubusercontent.com/18688696/171335169-71234734-7b12-49fb-8695-6440fc8ab0ba.png)

**Reply by janpio** [reply]
That is indeed weird.

In your Network panel in the developer tools you should be able to look at the content that is being returned for that request to figure out what your API is actually returning. (It might be something "function does not exist" or similar in theory, which would explain why there are not logs of unsuccessful requests - although that would be pretty weird.)

**Reply by jbarrasso** [reply]
I hear ya, yeah it's odd. Do you mean this tab? The response has no data to display

![image](https://user-images.githubusercontent.com/18688696/171473208-90a6f472-98f6-4183-b6b1-da6a6fa4b707.png)

**Reply by janpio** [reply]
That was indeed what I meant, but seems it is just a 404 with no content :/ 

You will need to figure out _why_ this function call returns a 404. Is the URL maybe just invalid in the context of the deployment? If not, why does the function execute but return a 404? Why can you not find any logs that would have more information about that 404?

All those are really more questions for Vercel though and not for Prisma unfortunately. It is highly unprobably that Prisma is the cause for this.

We are happy to help you further anyway of course, we have quite some experience deploying stuff to Vercel. Is your project public by any chance or can you share it with us?

**Reply by jbarrasso** [reply]
Definitely appreciate your help and time @janpio, our discussion is already proving to be pretty helpful. I had no luck in the vercel dicussion board unfortunately. I appreciate you becoming a part-time Vercel support member for me haha.

Yes my repo is public. Main will reproduce the issue.

The caveat is the api fetch that results in 404 will only happen on certain conditions:

1. you need a metamask wallet installed as a chrome extension.
AND
2.  your wallet address needs to own a specific NFT.

I think you'd be able to see the error in the Network tab by simply visiting any one of these URLs:
https://toadzgotchi-app-nu.vercel.app/api/users/0xb75F87261a1FAC3a86f8A48d55597A622BA3CC48
https://toadzgotchi-app-git-main-jbarrasso.vercel.app/api/users/0xb75F87261a1FAC3a86f8A48d55597A622BA3CC48
https://toadzgotchi-app-jbarrasso.vercel.app/api/users/0xb75F87261a1FAC3a86f8A48d55597A622BA3CC48

0xb75F87261a1FAC3a86f8A48d55597A622BA3CC48 is an Ethereum wallet that owns an NFT I am checking for

repo:
https://github.com/jbarrasso/toadzgotchi-app

**Reply by janpio** [reply]
To me the 404 looks like the default Next 404 that happens on any invalid URL like https://toadzgotchi-app-nu.vercel.app/foo/bar - so the serverless function behind `/api/users/[id]` is not actually triggered. So something is wrong with the configuration there that should make such requests responded to by https://github.com/jbarrasso/toadzgotchi-app/blob/main/pages/api/users/%5Bid%5D.tsx.

**Reply by jbarrasso** [reply]
Right, that makes sense. Especially since I was trying to configure the response headers within the serverless function (no effect). That's so strange though, what about this API call would prevent the function from being called? It's quite simple. We actually know it's being called because there is a failed request showing in the browser. The question is, like you said, why isn't the function being created/executed? I boiled it down pretty simply too for debugging purposes...

![image](https://user-images.githubusercontent.com/18688696/171776762-541947a6-9709-4613-a677-b99a78ea12a6.png)

I don't think it can get much simpler. Chalk this up to a bug? Not sure where to go from here...

All of my presets are set correctly within the Vercel dashboard.

Is there anything special I need to do to enable serverless functions?

**Reply by janpio** [reply]
That is now unfortunately in territory where I just don't know. This is something around Vercel and Next.js.

On the other hand, we have a testing project that uses Next.js and is deployed to Vercel: https://github.com/prisma/ecosystem-tests/tree/dev/platforms-serverless-vercel/vercel-with-nextjs The function we have there can be seen here: https://github.com/prisma/ecosystem-tests/blob/dev/platforms-serverless-vercel/vercel-with-nextjs/pages/api/index.js Seems it is accessible directly via `/api`. Maybe that helps...

**Comment by jbarrasso** [comment]
@nurul3101 I tried with and without Data Proxy and still not working :(

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
